### PR TITLE
rename Extensions to Stacks (design doc only)

### PR DESCRIPTION
**Pull request by displague**
Addresses the Extensions -> Stacks "design doc" concerns expressed in https://github.com/crossplaneio/crossplane/issues/571 

This name change throughout the design doc reflects the change in terminology used by the community so that newcomers to the community are not confused by the use of old terminology.

This PR does not tackle renaming code from `Extension*` to `Stack*`.   The updated design doc references these objects according to their existing names.

Additional issue links have been added to the open questions list and the design doc has been given a minor version bump.

### Checklist

I have:
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation], [examples], or [release notes].
- [ ] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

### CODE REVIEW

**Comment by hasheddan** [comment]
LGTM, I assume another PR will be needed after updating of the code references. Do you want to change doc name to `design-doc-stacks.md` in this PR?

File: design/design-doc-stacks.md [hunk 1]
```diff
@@ -1,15 +1,21 @@
-# Crossplane Extensions
+# Crossplane Stacks
 * Owner: Jared Watts (@jbw976)
 * Reviewers: Crossplane Maintainers
-* Status: Accepted, revision 1.0
+* Status: Accepted, revision 1.1
 
-This document aims to provide details about the experience and implementation for Crossplane “extensions”, which can add new functionality/support, types, and controllers to Crossplane.
+This document aims to provide details about the experience and implementation for Crossplane “Stacks”, which can add new functionality/support, types, and controllers to Crossplane.
```
Comment (negz) [inline design/design-doc-stacks.md]: Should Stacks remain in quotations given it's almost as much a brand as a concept?

File: design/design-doc-stacks.md [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+* 1.1
+  * Renamed Extensions concept to Stacks (`Extension` code references are unaffected) [#571](https://github.com/crossplaneio/crossplane/issues/571)
+  * Added additional Questions and Open Issues
 
 ## Experience
 
 The core experience for consuming new functionality in Crossplane is composed of 2 steps:
 
-1. Create an extension request for the name of the extension or one of the CRDs that it owns
+1. Create an extension request for the name of the Crossplane Stack or one of the CRDs that it owns
```
Comment (negz) [inline design/design-doc-stacks.md]: Is "extension request" still the preferred terminology for installing a Stack?

File: design/design-doc-stacks.md [hunk 3]
```diff
@@ -103,32 +109,32 @@ status:
 
 ## `Extension` CRD
 
-The `Extension` CRD serves as a record of an installed extension (a custom controller and its CRDs).
+The `Extension` CRD serves as a record of an installed Stack (a custom controller and its CRDs).
```
Comment (negz) [inline design/design-doc-stacks.md]: Do we plan to rename the `Extension` CRD to `Stack` at some point, or will this remain unchanged?

Comment (negz) [reply design/design-doc-stacks.md]: I noticed we do rename it to `Stack` below, so I think we need to make sure these are consistent.

File: design/design-doc-stacks.md (hunk 2 above)
Comment (displague) [reply design/design-doc-stacks.md]: A design-doc-only change leaves some of these questions open, I feel.

This set of changes reflects the transition in terminology as a precursor to any code changes.

I don't want to confuse anyone reading this and then comparing notes against the code where types and kinds are still called `Extension`.  It was my intent to leave `Extension` and `ExtensionRequest` references alone for now, even if it is only for a few days/weeks.

File: design/design-doc-stacks.md [hunk 4]
```diff
@@ -103,32 +109,32 @@ status:
...
 
 `Extension` CRD instances can also be created directly by a user without any knowledge of packages at all.
-They can directly create any CRDs that their extension requires and then create an `Extension` CRD instance that describes their extension, its custom controller, etc.
-The Extension Manager will see this new instance and take the steps necessary to ensure the custom controller is running in the cluster and the extension’s functionality is available.
+They can directly create any CRDs that their Stack requires and then create an `Extension` CRD instance that describes their Stack, its custom controller, etc.
+The Stack Manager will see this new instance and take the steps necessary to ensure the custom controller is running in the cluster and the Stack’s functionality is available.
 
 ```yaml
 apiVersion: extensions.crossplane.io/v1alpha1
 kind: Extension
```
Comment (negz) [inline design/design-doc-stacks.md]: Without much context on this change, `Extension.stack.crossplane.io` and `ExtensionRequest.stack.crossplane.io` may be good names if we decide to rename the CRDs. Similar to how the “workload” functionality is implemented using `KubernetesApplication.workload.crossplane.io`

Comment (displague) [reply design/design-doc-stacks.md]: I'll keep this in mind when creating (or reviewing) the follow-up PR to rename the code.

File: design/design-doc-stacks.md (hunk 1 above)
Comment (displague) [reply design/design-doc-stacks.md]: I considered dropping the quotes too.  Consider them dropped.

File: design/design-doc-stacks.md (hunk 2 above)
Comment (displague) [reply design/design-doc-stacks.md]: I'll reword this to "installation request" to disambiguate it from the legacy Extensions while preserving the same meaning.

File: design/design-doc-stacks.md (hunk 3 above)
Comment (displague) [reply design/design-doc-stacks.md]: I've corrected the inadvertent `Kind: Stack`. Thanks.

A follow-up PR will pick up code renaming of the CRD (or choose not to, a benefit of this PR being separate and less aggressive).
